eth/tracers/logger/logger.go

```diff
@@ -177,14 +177,14 @@ func (s *StructLog) toLegacyJSON() json.RawMessage {
 	if len(s.ReturnData) > 0 {
 		msg.ReturnData = hexutil.Bytes(s.ReturnData).String()
 	}
-	if s.Memory != nil {
+	if len(s.Memory) > 0 {
```

Comment:
This change does fix the issue, although I have no idea why the empty memory was omitted occasionally. But I think we can definitely go with this direction.

Reply:
Btw, in line 182, it's wrong. If the length of memory is 1, then the for loop will never be executed

Given the fact that memory expansion always happens in 32-byte (word-sized) chunks, so should be fine to keep it